adam_core/coordinates/members.py

```diff
@@ -142,6 +194,30 @@ def keplerian(self):
 
         return self._keplerian
 
+    @keplerian.setter
+    def keplerian(self, coords: KeplerianCoordinates):
+        if "KeplerianCoordinates" not in self.__allowed_coordinate_types:
```

Comment:
Great question. This is purely to prevent users from doing something silly. The upcoming observations/ephemeris classes will also have CoordinateMembers but allowing those observations to be expressed as Keplerian or Cometary doesn't make much sense. I wanted to make sure the user couldn't convert say optical spherical observations to Keplerian and interpret them as actual orbits because they wouldn't be.